File: objective-c/Touch3D/Touch3D/Handlers/MainHandler.m
//
//  MainHandler.m
//  Touch3D
//
//  Created by Alexandr Kirilov on 09/06/2019.
//  Copyright © 2019 none. All rights reserved.
//

#import "MainHandler.h"

@implementation MainHandler

+(void) mDoExit {
	
    UIAlertController *oAlert = [UIAlertController
    	alertControllerWithTitle:@"Exit"
    	message:@"Are sure about to exit application?"
    	preferredStyle:UIAlertControllerStyleAlert
    ];
    UIAlertAction *oActionYes = [UIAlertAction
    	actionWithTitle:@"Yes"
    	style:UIAlertActionStyleDefault
    	handler:^(UIAlertAction *action){
    		exit(0);
		}
    ];
    UIAlertAction *oActionNo = [UIAlertAction
    	actionWithTitle:@"Cancel"
    	style:UIAlertActionStyleDefault
    	handler:^(UIAlertAction *action){}
    ];
    [oAlert addAction:oActionNo];
    [oAlert addAction:oActionYes];
    [[UIApplication sharedApplication].keyWindow.rootViewController presentViewController:oAlert
    	animated:YES
    	completion:nil
    ];
}

+(UIColor *) mCreateRandomColor {
	
	return [UIColor
		colorWithRed: (arc4random() % 255) / 255.0
		green: (arc4random() % 255) / 255.0
		blue: (arc4random() % 255) / 255.0
		alpha: 1.0
	];
}

@end

File: swift/Collection View/v1/Collection View/Handlers/MainHandler.swift
//
//  MainHandler.swift
//  Collection View
//
//  Created by Alexandr Kirilov on 19/07/2019.
//  Copyright © 2019 Arboreus. All rights reserved.
//

import UIKit

class MainHandler: NSObject {

	static func mDoExit() {

		let oAlertController: UIAlertController = UIAlertController(
			title: "Exit",
			message: "Application is about exit. Are you sure?",
			preferredStyle: .alert
		);
		oAlertController.addAction(UIAlertAction.init(
			title: "Exit", style: .default, handler: {_ in
			print("Application is about to exit");
			exit(0);
		}));
		oAlertController.addAction(UIAlertAction.init(
			title: "Cancel", style: .cancel, handler: {_ in
			print("Application exit canceled")
		}));
		UIApplication.shared.keyWindow?.rootViewController?.present(
			oAlertController, animated: true
		);
	}
}
